Sources/AsyncHTTPClient/ConnectionPool/HTTP2/HTTP2IdleHandler.swift

```diff
@@ -0,0 +1,224 @@
+                preconditionFailure("Invalid state: \(self.state)")
+
+            case .connected:
+                let maxStreams = settings.last(where: { $0.parameter == .maxConcurrentStreams })?.value ?? 100
+                self.state = .active(openStreams: 0, maxStreams: maxStreams)
+                return .notifyConnectionNewSettings(settings)
+
+            case .active(openStreams: let openStreams, maxStreams: let maxStreams):
+                if let newMaxStreams = settings.last(where: { $0.parameter == .maxConcurrentStreams })?.value, newMaxStreams != maxStreams {
+                    self.state = .active(openStreams: openStreams, maxStreams: newMaxStreams)
+                    return .notifyConnectionNewSettings(settings)
+                }
+                return .nothing
+
+            case .closing:
+                return .nothing
+            }
+        }
+
+        mutating func goAwayReceived() -> Action {
+            switch self.state {
+            case .initialized, .closed:
+                preconditionFailure("Invalid state")
+            case .connected:
+                self.state = .closing(openStreams: 0, maxStreams: 0)
+                return .notifyConnectionGoAwayReceived(close: true)
+
+            case .active(let openStreams, let maxStreams):
+                self.state = .closing(openStreams: openStreams, maxStreams: maxStreams)
+                return .notifyConnectionGoAwayReceived(close: openStreams == 0)
+
+            case .closing:
+                return .notifyConnectionGoAwayReceived(close: false)
+            }
+        }
+
+        mutating func closeEventReceived() -> Action {
+            switch self.state {
+            case .initialized:
+                preconditionFailure("")
+
+            case .connected:
+                self.state = .closing(openStreams: 0, maxStreams: 0)
+                return .close
+
+            case .active(let openStreams, let maxStreams):
+                if openStreams == 0 {
+                    self.state = .closed
+                    return .close
+                }
+
+                self.state = .closing(openStreams: openStreams, maxStreams: maxStreams)
+                return .nothing
+
+            case .closed, .closing:
+                return .nothing
+            }
+        }
+
+        mutating func streamCreated() -> Action {
+            guard case .active(var openStreams, let maxStreams) = self.state else {
+                preconditionFailure("Invalid state")
+            }
+
+            openStreams += 1
+            assert(openStreams <= maxStreams)
+
+            self.state = .active(openStreams: openStreams, maxStreams: maxStreams)
+            return .nothing
+        }
+
+        mutating func streamClosed() -> Action {
+            guard case .active(var openStreams, let maxStreams) = self.state else {
+                preconditionFailure("Invalid state")
+                // TODO: What happens, if we received a go away?!??!
```

Comment:
We should handle this in a similar way to active but we should close the connection when `openStreams` drops to zero.